register-form/src/components/RegForm.vue

```diff
@@ -0,0 +1,186 @@
+    <br>  
+
+
+    <label>Email:</label><br>
+    <input :class="{invalid: !valids.email & !firstTime}" type="text" v-model="inputs.email">
+    <br>
+    <span class="error" v-show="!valids.email & !firstTime">*Email must have the form: 'X@X.X'.</span>
+    <br>
+
+    <label>Phone number:</label><br>
+    <input :class="{invalid: !valids.phoneNum & !firstTime}" type="text" v-model="inputs.phoneNum">
+    <br>
+    <span class="error" v-show="!valids.phoneNum & !firstTime">*Phone number must have 10 digits.</span>
+    <br>
+    
+    <p>Chose your power animal:</p>
+    <label v-for="img in images" :key="img.val" :class="{invalidAvatar: !valids.avatar & !firstTime}">
+      <input type="radio" :value="img.val" v-model="inputs.avatar">
+      <img class="radio-img" :src="img.src" :alt="img.val"> 
+    </label>
+    <br>
+    <span class="errorAvatar" v-show="!valids.avatar & !firstTime">*Must chose an avatar.</span>
+    <br>
+
+    <input type="button" class="submit" value="Submit" @click="submitForm()">
+    <br><br>
+
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'RegForm',
+  data(){
+    return {
+
+      inputs: {firstName: "", lastName: "", gender: "", 
+                  email: "", phoneNum: "", avatar: ""},
+
+      images: [{src: require("..\\assets\\elk.svg"), val: "Elk"}, 
+              {src: require("..\\assets\\Gorilla.svg"), val: "Gorilla"},
+              {src: require("..\\assets\\snake.svg"), val: "Snake"}],
+
+      firstTime: true
+    }
+  },
+  computed:{
+
+    isFirstNameValid: function(){
+      return this.inputs.firstName.length > 0
+    },
+    isLastNameValid: function(){
+      return this.inputs.lastName.length > 0
+    },
+    isGenderValid: function(){
+      return this.inputs.gender.length > 0
+    },
+    isEmailValid: function(){
+      let regexEmail =/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;
+      if (this.inputs.email.match(regexEmail)) {
+        return true; 
+      } 
+      return false; 
+    },
+    isPhoneNumberValid: function(){
+      let phoneno = /^\d{10}$/;
+      if(this.inputs.phoneNum.match(phoneno)){
+        return true
+      }
+      return false
+    },
+
+    isAvatarValid: function(){
+      return this.inputs.avatar.length > 0
+    },
+
+    valids: function(){
+            return {firstName: this.isFirstNameValid, lastName: this.isLastNameValid,
+               gender: this.isGenderValid, email: this.isEmailValid, 
+               phoneNum: this.isPhoneNumberValid, avatar: this.isAvatarValid}
+    },
+    
+  },
+  methods:{
+    submitForm: function(){
+      this.firstTime = false
+      if(this.allValid())
+        console.log(this.inputs)
+    },
+    allValid : function(){
+      let areAllValid = true
+      for(let key in this.valids){
+        areAllValid = areAllValid & this.valids[key]
+      }
+      return areAllValid
+    }
+  }
+}
+</script>
+
+<style scoped>
+
+  input, select{
```

Comment:
i did it because the reason you mentioned and because i used `<style scoped>` which prevent from an inside componnent to take any of this componnent CSS styles.